tests/app/main/test_services.py

```diff
@@ -253,12 +253,69 @@ def test_should_view_public_service_with_correct_message(
             res.get_data(as_text=True)
         )
 
-        # service removed notification banner shouldn't be there either
+        # service removed notification banner shouldn't be there
         self.assert_not_in_strip_whitespace(
             '<h2>This service was removed on Monday 23 March 2015</h2>',
             res.get_data(as_text=True)
         )
 
+        # service updated message shouldn't be there either
+        self.assert_not_in_strip_whitespace(
+            "The changes are now live on the Digital Marketplace",
+            res.get_data(as_text=True)
+        )
+
+        self._post_remove_service(service_should_be_modifiable=True)
+
+    def test_should_view_public_service_with_update_message(
+            self, data_api_client, fixture_framework_slug_and_name
+    ):
+        self.login()
+        self._get_service(data_api_client, fixture_framework_slug_and_name, service_status='published')
+
+        # this is meant to emulate a "service updated" message
+        with self.client.session_transaction() as session:
+            session['_flashes'] = [
+                ('message', 'Foo Bar 123 321'),
+            ]
+
+        res = self.client.get('/suppliers/services/123')
+        assert_equal(res.status_code, 200)
+
+        assert_true(
+            'Service name 123' in res.get_data(as_text=True)
+        )
+
+        # first message should be there
+        self.assert_in_strip_whitespace(
+            'Remove this service',
+            res.get_data(as_text=True)
+        )
+
+        # removal confirmation message should not have been triggered
+        self.assert_not_in_strip_whitespace(
+            'Are you sure you want to remove your service?',
+            res.get_data(as_text=True)
+        )
+
+        # service removed message should not have been triggered
+        self.assert_not_in_strip_whitespace(
+            'Service name 123 has been removed.',
+            res.get_data(as_text=True)
+        )
+
+        # service removed notification banner shouldn't be there
+        self.assert_not_in_strip_whitespace(
+            '<h2>This service was removed on Monday 23 March 2015</h2>',
```

Comment:
Where does the date here come from? I guess it must be in the fixture somehow but I can't see where?  

Still, maybe just asserting not `<h2>This service was removed on` would cover more possible failure cases?